### PR TITLE
Set platform in sentry frames for better raw stacktrace representation

### PR DESCRIPTION
**Pull request by buenaflor**
## :scroll: Description

Sets either dart or javascript platform to frames for raw stacktrace representation

## :bulb: Motivation and Context

Issue: https://github.com/getsentry/sentry-dart/issues/2040

## :green_heart: How did you test it?
Unit test

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [ ] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by kahest** [comment]
@buenaflor changes look good 👍 now that https://github.com/getsentry/sentry/pull/74715 is merged we can test end-to-end

**Comment by buenaflor** [comment]
image

looks good, still gotta fix the tests

**Review by markushi — APPROVED** [review]
LGTM!

**Review by kahest — APPROVED** [review]
awesome, thanks!

File: dart/lib/src/sentry_stack_trace_factory.dart [hunk 1]
```diff
@@ -103,6 +103,7 @@ class SentryStackTraceFactory {
       // least we get an indication something's wrong and are able to fix it.
     }
 
+    final platform = _options.platformChecker.isWeb ? 'javascript' : 'dart';
```
Comment (ekuleshov) [inline dart/lib/src/sentry_stack_trace_factory.dart:106]: does this mean we aren't going to see the Dart stack traces coming from the web platform?

Comment (buenaflor) [reply dart/lib/src/sentry_stack_trace_factory.dart:106]: this is just about the raw formatting, you will still be able to see everything as before

Comment (ekuleshov) [reply dart/lib/src/sentry_stack_trace_factory.dart:106]: before it wasn't showing proper Dart-formatted stack traces on any platform.

Need to be able to see the dart stacktraces on desktop, mobile and web targets.

Also, this seem to be SDK change? The correct stack traces won't be shown on Sentry's web UI until the apps are updated and new errors/stacktraces are pushed?

Comment (buenaflor) [reply dart/lib/src/sentry_stack_trace_factory.dart:106]: see my comment  here about what we can do if you want it to work on existing issues:

https://github.com/getsentry/sentry-dart/issues/2040#issuecomment-2250159494

let's continue the discussion over there

**Pull request merged**